pkg/payload/task_graph_test.go

```diff
@@ -711,27 +783,34 @@ func TestRunGraph(t *testing.T) {
 		}
 		return arr
 	}
+
 	tests := []struct {
-		name     string
-		nodes    []*TaskNode
-		parallel int
-		sleep    time.Duration
-		errorOn  func(t *testing.T, name string, ctx context.Context, cancelFn func()) error
+		name  string
+		nodes []*TaskNode
+		// maxParallelism is a RunGraph parameter affecting how many nodes are processed in parallel
+		maxParallelism int
+		// sleep is a duration to sleep in each task to simulate work
+		sleep time.Duration
+		// callbacks allows providing a method to be called for a specific task, to simulate work, return error or trigger
+		// cancellations
+		callbacks map[string]callbackFn
```

Comment:
possibly doc the `*` special-case here, from `for _, callbackKey := range []string{task.Manifest.OriginalFilename, "*"}`?  An alternative would be to have the keys be regular expressions, but :shrug:, you already have it working your way, and "task name or `*` wildcard" isn't complicated to explain.

Reply:
/hold in case you want to do this^

Feel free to lift the hold without making changes, if you're fine as things stand.

Reply:
Good point, will do

Reply:
Addressed

/hold cancel